components/indexer.py

```diff
@@ -5,15 +5,42 @@ class Indexer:
     indexer_motors: list
     indexer_switches: list
 
-    def __init__(self):
-        pass
+    def on_enable(self) -> None:
+        self.indexing = True
 
-    def setup(self):
-        pass
+    def execute(self) -> None:
+        if self.indexing:
+            motor_states = [True] * len(self.indexer_motors)
+            switch_results = [switch.get() for switch in self.indexer_switches]
+            if not switch_results[
+                0
+            ]:  # Test the back switch (because all others require 2 switches)
+                motor_states[0] = False
 
-    def on_enable(self):
-        for motor in self.indexer_motors:
-            motor.stopMotor()
+            for i in range(
+                1, len(motor_states)
+            ):  # Disable motor if switch and previous switch are pressed
+                if not switch_results[i] and not switch_results[i - 1]:
+                    motor_states[i] = False
+
+            for i in range(len(motor_states)):  # Set all motors to required states
+                if motor_states[i]:
```

Comment:
Why are you looping over `range(len(motor_states))` then accessing `motor_states` with that value?

Reply:
Because we then use that value to access indexer_motors.